### ISSUE TITLE
AttributeError: module 'mistune' has no attribute 'BlockGrammar'

### ISSUE THREAD

**Issue by Davidy22** [jupyter/notebook#6531]
**Describe the bug**
There is an error, "AttributeError: module 'mistune' has no attribute 'BlockGrammar'", caused by the mistune library changing their whole structure and the intermediate dependency m2r not being updated or pinning deps. There is a library [m2r2](https://github.com/CrossNox/m2r2) that aims to be maintained and fix that issue because m2r appears to be unresponsive, and switching to that I believe should resolve the issue.

**To Reproduce**

1. Try to open a notebook

**Expected behavior**
Notebook opens

**Desktop (please complete the following information):**

- OS: Fedora 36
- Browser agnostic

23:13:50❯ jupyter --version
Selected Jupyter core packages...
IPython          : 8.4.0
ipykernel        : 6.15.0
ipywidgets       : 7.7.1
jupyter_client   : 7.3.4
jupyter_core     : 4.10.0
jupyter_server   : not installed
jupyterlab       : not installed
nbclient         : 0.6.6
nbconvert        : 6.5.0
nbformat         : 5.4.0
notebook         : 6.4.12
qtconsole        : 5.3.1
traitlets        : 5.3.0

**Comment by RRosio** [comment]
Hi @Davidy22 thank you for submitting this issue. We found [this issue](https://github.com/jupyter/nbconvert/issues/1787) in nbconvert that seems to be related. It seems like in nbconvert version 7.0.0 this may be fixed, but I'm tagging @blink1073 in case Steve can provide more information.

**Comment by blink1073** [comment]
I believe this is related to https://github.com/sphinx-contrib/openapi/issues/123.  We don't use `mr2` directly in any Jupyter libraries as far as I know, but if it is pulled in by a dependency, then you need to add `mistune<1` to your requirements file, or manually install that version.

**Comment by grisaitis** [comment]
> It seems like in nbconvert version 7.0.0 this may be fixed, but I'm tagging @blink1073 in case Steve can provide more information.

I'm experiencing this with nbconvert 7.0.0 :/

```

**Comment by Davidy22** [comment]
Yeah, it's an intermediate dependency. I was already able to get opening notebooks to work by downgrading mistune but there'll probably be other people impacted by the cascading dependency issues flowing from m2r. Actually, I can grab the stack trace when mistune is the latest version:

```
    Traceback (most recent call last):
      File "/home/davidos/.local/lib/python3.10/site-packages/tornado/web.py", line 1713, in _execute
        result = await result
      File "/home/davidos/.local/lib/python3.10/site-packages/tornado/gen.py", line 782, in run
        yielded = self.gen.send(value)
      File "/home/davidos/.local/lib/python3.10/site-packages/notebook/notebook/handlers.py", line 94, in get
        self.write(self.render_template('notebook.html',
      File "/home/davidos/.local/lib/python3.10/site-packages/notebook/base/handlers.py", line 511, in render_template
        return template.render(**ns)
      File "/home/davidos/.local/lib/python3.10/site-packages/jinja2/environment.py", line 1301, in render
        self.environment.handle_exception()
      File "/home/davidos/.local/lib/python3.10/site-packages/jinja2/environment.py", line 936, in handle_exception
        raise rewrite_traceback_stack(source=source)
      File "/home/davidos/.local/lib/python3.10/site-packages/notebook/templates/notebook.html", line 1, in top-level template code
        {% extends "page.html" %}
      File "/home/davidos/.local/lib/python3.10/site-packages/notebook/templates/page.html", line 154, in top-level template code
        {% block header %}
      File "/home/davidos/.local/lib/python3.10/site-packages/notebook/templates/notebook.html", line 115, in block 'header'
        {% for exporter in get_frontend_exporters() %}
      File "/home/davidos/.local/lib/python3.10/site-packages/notebook/notebook/handlers.py", line 23, in get_frontend_exporters
        from nbconvert.exporters.base import get_export_names, get_exporter
      File "/home/davidos/.local/lib/python3.10/site-packages/nbconvert/__init__.py", line 3, in <module>
        from . import filters, postprocessors, preprocessors, writers
      File "/home/davidos/.local/lib/python3.10/site-packages/nbconvert/filters/__init__.py", line 8, in <module>
        from .markdown import *
      File "/home/davidos/.local/lib/python3.10/site-packages/nbconvert/filters/markdown.py", line 13, in <module>
        from .markdown_mistune import markdown2html_mistune
      File "/home/davidos/.local/lib/python3.10/site-packages/nbconvert/filters/markdown_mistune.py", line 37, in <module>
        class MathBlockGrammar(mistune.BlockGrammar):
    AttributeError: module 'mistune' has no attribute 'BlockGrammar'
```

I don't remember why I pointed the finger at m2r, but it seems like the last offending dependency before the invalid usage of the overhauled mistune is nbconvert. Looks like another jupyter repo, is there a way to move an issue or should I recreate or is this issue here sufficient?

**Comment by blink1073** [comment]
Pinning mistune will work in all cases, but the issue in nbconvert has already been fixed through the 7.0 release.  https://github.com/jupyter/nbconvert/issues/1842

**Comment by Davidy22** [comment]
Oh, I guess that makes this issue moot then. How'd my pip jupyter install give me nbconvert 6.5? Well, shouldn't be an issue going forward for new installs then I think.
